src/luarocks/fs/lua.lua

```diff
@@ -785,6 +770,40 @@ function fs_lua.get_permissions(file)
    return posix.stat(file, "mode")
 end
 
+--- Create a temporary directory.
+-- @param name string: name pattern to use for avoiding conflicts
+-- when creating temporary directory.
+-- @return string or (nil, string): name of temporary directory or (nil, error message) on failure.
+function fs_lua.make_temp_dir(name)
+   assert(type(name) == "string")
+   name = dir.normalize(name)
+
+   local template = (os.getenv("TMPDIR") or "/tmp") .. "/luarocks_" .. name:gsub(dir.separator, "_") .. "-XXXXXX"
+   local temp_dir, err = posix.mkdtemp(template)
+   if temp_dir then
+      return temp_dir
+   else
+      return nil, err
+   end
+end
+
+--- Create and return the name of a temporary file.
+-- @return string or (nil, string): name of temporary file or (nil, error message) on failure.
+function fs_lua.tmpname()
```

Comment:
Hi! Thanks for contributing! 
I'm confused about this one: in which way is this better than just using `os.tmpname()`? Just the fact that we add `luarocks_` to the name?